File: tournaments/decipher/decipher.cpp
std::string decipher(std::string cipher)
{
	std::string m;
	std::string s;
	for (char ch : cipher)
	{
		s += ch;
		if (97 <= std::stoi(s) && std::stoi(s) <= 122)
		{
			m += std::stoi(s);
			s = "";
		}
	}
	return m;
}

File: tournaments/decipher/decipher.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub decipher {
    my ($cipher) = @_;

    my @store = split('', $cipher);
    my $result = '';
    my $x = '';

    foreach (@store) {
        $x .= $_;
        if (97 <= int($x) && 122 >= int($x)) {
            $result .= chr($x);
            $x = '';
        }
    }

    return $result;
}
